internal/stage8.go

```diff
@@ -16,39 +16,67 @@ import (
 func testRun(stageHarness *test_case_harness.TestCaseHarness) error {
 	logger := stageHarness.Logger
 	shell := shell_executable.NewShellExecutable(stageHarness)
-	randomCode := getRandomString()
-	randomName := getRandomName()
-	randomExecutableName := "custom_exe_" + strconv.Itoa(random.RandomInt(1000, 9999))
+
+	randomCode1 := getRandomString()
+	randomCode2 := getRandomString()
+
+	randomName1 := getRandomName()
+	randomName2 := getRandomName()
+	randomName3 := getRandomName()
+
+	randomExecutableName1 := "custom_exe_" + strconv.Itoa(random.RandomInt(1000, 9999))
+	randomExecutableName2 := "custom_exe_" + strconv.Itoa(random.RandomInt(1000, 9999))
+
 	_, err := SetUpCustomCommands(stageHarness, shell, []CommandDetails{
-		{CommandType: "signature_printer", CommandName: randomExecutableName, CommandMetadata: randomCode},
+		{CommandType: "signature_printer", CommandName: randomExecutableName1, CommandMetadata: randomCode1},
+		{CommandType: "signature_printer", CommandName: randomExecutableName2, CommandMetadata: randomCode2},
 	}, true)
 	if err != nil {
 		return err
 	}
-	logAvailableExecutables(logger, []string{randomExecutableName})
+	logAvailableExecutables(logger, []string{randomExecutableName1, randomExecutableName2})
 	asserter := logged_shell_asserter.NewLoggedShellAsserter(shell)
 
 	if err := asserter.StartShellAndAssertPrompt(true); err != nil {
 		return err
 	}
 
-	command := []string{
-		randomExecutableName, randomName,
+	command1 := []string{
+		randomExecutableName1, randomName1,
 	}
 
 	testCase := test_cases.CommandWithMultilineResponseTestCase{
-		Command: strings.Join(command, " "),
+		Command: strings.Join(command1, " "),
 		MultiLineAssertion: assertions.NewMultiLineAssertion([]string{
-			fmt.Sprintf("Program was passed %d args (including program name).", len(command)),
-			fmt.Sprintf("Arg #0 (program name): %s", command[0]),
-			fmt.Sprintf("Arg #1: %s", command[1]),
-			fmt.Sprintf("Program Signature: %s", randomCode),
+			fmt.Sprintf("Program was passed %d args (including program name).", len(command1)),
+			fmt.Sprintf("Arg #0 (program name): %s", command1[0]),
```

Comment:
@andy1li let's add a bit of randomness here - select 1-3 args (random) and execute the command two times